File: HARD/F-hard02-maximum-gap.py
# Resolução do problema https://leetcode.com/problems/maximum-gap/
# Dificuldade: Difícil


def maximumGap(nums):
    if len(nums) < 2:
        return 0
    lista = []
    nums = sorted(nums)
    for i in range(len(nums) - 1):
        a = abs(nums[i + 1] - nums[i])
        lista.append(a)
    maior = max(lista)
    return maior


print(maximumGap(nums = [3,6,9,1]))

File: HARD/F-hard02-maximum-gap.js
// Resolução do problema https://leetcode.com/problems/maximum-gap
// Dificuldade: Fácil


var maximumGap = function(nums) {
    if (nums.length < 2) {
        return 0
    }
    let lista = []
    nums = nums.sort(function(a, b){return a - b})
    for (let i = 0; i < nums.length - 1; i++) {
        a = nums[i + 1] - nums[i]
        lista.push(a)
    }
    let maior = lista.reduce(function(a, b) {
        return Math.max(a, b)
    }, -Infinity)
    console.log(nums)
    console.log(lista)
    ;
    return maior
};


console.log(maximumGap(nums = [1, 3, 100]))
console.log(maximumGap(nums = [10]))